From 124ab798e65b6c95a8486f6f6af9bdf69b11e1bf Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ond=C5=99ej=20Va=C5=A1=C3=ADk?= <ovasik@redhat.com>
Date: Fri, 11 Oct 2013 14:44:53 +0200
Subject: [PATCH] cp: correct error message for invalid arguments of '--no-preserve'

* src/cp.c (decode_preserve_arg) :
Correct error message for invalid arguments of '--no-preserve'.
Reported by M.Vadkerti in rhbz #1018206
---
 src/cp.c |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/src/cp.c b/src/cp.c
index e235b32..7bc8630 100644
--- a/src/cp.c
+++ b/src/cp.c
@@ -854,7 +854,8 @@ decode_preserve_arg (char const *arg, struct cp_options *x, bool on_off)
         *comma++ = 0;
 
       /* process S.  */
-      val = XARGMATCH ("--preserve", s, preserve_args, preserve_vals);
+      val = XARGMATCH (on_off ? "--preserve" : "--no-preserve",
+                       s, preserve_args, preserve_vals);
       switch (val)
         {
         case PRESERVE_MODE:
-- 
1.7.1

